pkg/controllers/provisioning/scheduling/scheduler.go

```diff
@@ -445,10 +446,17 @@ func (s *Scheduler) updateCachedPodData(p *corev1.Pod) {
 		Requests:           resources.RequestsForPods(p),
 		Requirements:       requirements,
 		StrictRequirements: strictRequirements,
+		IsDRAEnabled:       pod.HasDRARequirements(p),
 	}
 }
 
 func (s *Scheduler) add(ctx context.Context, pod *corev1.Pod) error {
+	// Check if pod has DRA requirements - if so, return immediately as DRA is not yet supported
+	if s.cachedPodData[pod.UID].IsDRAEnabled {
+		log.FromContext(ctx).V(1).WithValues("Pod", klog.KObj(pod)).Info("skipping pod scheduling, Dynamic Resource Allocation (DRA) is not yet supported by Karpenter")
```

Comment:
Is there a reason we're just logging rather than returning an error here? No error suggests that we successfully scheduled a pod. Returning an error here wouldn't short-circuit the scheduler, and it has the value of ensuring that we include the error in the scheduling result. 

One thing we may want to consider is wrapping it in a special type of error (e.g. `DRAError`) and not inserting the pod back into the scheduling queue if the error is of that type. This would be similar to how we don't relax constraints if the error is of type `ReservedCapacityError`.